File: problems/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0, words = 0, sentences = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters += 1;
        }
        else if (text[i] == ' ')
        {
            words += 1;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }
    words = words + 1;
    float L = 100 * letters / (float) words;
    float S = 100 * sentences / (float) words;

    float G = 0.0588 * L - 0.296 * S - 15.8;
    if (G < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        int G1 = round(0.0588 * L - 0.296 * S - 15.8);
        if (G1 > 16)
        {
            printf("Grade 16+\n");
        }
        else
        {
            printf("Grade %i\n", G1);
        }
    }
}

File: problems/readability.py
import cs50

text = cs50.get_string("Text: ")

letters = sentences = words = 0
for char in text:
    if char.isalpha():
        letters += 1
    if char.isspace():
        words += 1
    if char in ['!', '.', '?']:
        sentences += 1

words += 1

L = 100.0 * letters / words
S = 100.0 * sentences / words
G = int(round(0.0588 * L - 0.296 * S - 15.8))
if G < 1:
    print("Before Grade 1")
elif G >= 16:
    print("Grade 16+")
else:
    print(f"Grade {G}")